pkg/loop/telem.go

```diff
@@ -30,6 +55,73 @@ func SetupTelemetry(registerer prometheus.Registerer) GRPCOpts {
 	return GRPCOpts{DialOpts: dialOptions(registerer), NewServer: newServerFn(registerer)}
 }
 
+// SetupTracing initializes open telemetry with the provided config.
+// It sets the global trace provider and opens a connection to the configured collector.
+func SetupTracing(config TracingConfig) {
+	log, err := logger.New()
+	if err != nil {
+		panic(err)
+	}
+
+	ctx := context.Background()
+	ctx, cancel := context.WithTimeout(ctx, 5 * time.Second)
+
+	// Enough to shutdown the underlying connection since DialContext is used in blocking mode
+	defer cancel()
+	conn, err := grpc.DialContext(ctx, config.CollectorTarget,
+		// Note the use of insecure transport here. TLS is recommended in production.
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		// safe usage in this case https://github.com/grpc/grpc-go/blob/master/Documentation/anti-patterns.md#dialing-in-grpc
+		grpc.WithBlock(),
+	)
+	if err != nil {
+		log.Errorf("Connecting to OTEL collector %w failed: %v", config.CollectorTarget, err)
```

Comment:
Should this return?
```suggestion
		log.Errorf("Connecting to OTEL collector %w failed: %w", config.CollectorTarget, err)
		return
```

Reply:
Previously, no. With the new API, I think that's the right approach.